src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/HarborSelection.tsx

```diff
@@ -48,78 +54,65 @@ export const HarborSelection = forwardRef<
         >
           {t(PurchaseOverviewTexts.stopPlaces.harborSelection.select.text)}
         </ThemeText>
+
         <Section accessible={false}>
-          <GenericClickableSectionItem
-            ref={harborInputSectionItemRef}
-            accessible={true}
-            accessibilityRole="button"
-            accessibilityLabel={t(
-              PurchaseOverviewTexts.stopPlaces.harborSelection.from.a11yLabel(
-                fromHarbor?.name,
-              ),
-            )}
-            accessibilityHint={t(
-              PurchaseOverviewTexts.stopPlaces.harborSelection.from.a11yHint,
-            )}
-            onPress={() =>
-              onSelect({
-                fareProductTypeConfig,
-                preassignedFareProduct,
-              })
-            }
-            testID="selectHarborsButton"
-          >
-            <>
-              <View style={styles.fromHarbor}>
-                <ThemeText
-                  color="secondary"
-                  type="body__secondary"
-                  style={styles.toFromLabel}
-                >
-                  {t(PurchaseOverviewTexts.fromToLabel.from)}
-                </ThemeText>
-                <HarborLabel harbor={fromHarbor} />
-              </View>
-            </>
-          </GenericClickableSectionItem>
-          <GenericClickableSectionItem
-            accessible={true}
-            accessibilityState={{disabled: !fromHarbor}}
-            accessibilityRole="button"
-            accessibilityLabel={t(
-              PurchaseOverviewTexts.stopPlaces.harborSelection.to.a11yLabel(
-                toHarbor?.name,
-              ),
-            )}
-            accessibilityHint={
-              fromHarbor
-                ? t(
-                    PurchaseOverviewTexts.stopPlaces.harborSelection.to
-                      .a11yHint,
-                  )
-                : undefined
-            }
-            onPress={() =>
-              fromHarbor &&
-              onSelect({
-                fromHarbor,
-                fareProductTypeConfig,
-                preassignedFareProduct,
-              })
-            }
-            testID="selectHarborsButton"
-          >
-            <View style={styles.toHarbor}>
-              <ThemeText
-                color={!!fromHarbor ? 'secondary' : 'disabled'}
-                type="body__secondary"
-                style={styles.toFromLabel}
+          {([FROM, TO] as (typeof FROM | typeof TO)[]).map((fromOrTo) => {
```

Comment:
As alternative, you could create an internal component called `HarborSectionItem` that get as parameter a style, and if is disabled or not. The call it is twice like: 
```
<HarborSectionItem style={styles.fromHarbor}>
<HarborSectionItem style={styles.toHarbor} enabled={!fromHarbor}>
```

it could be internally to do not lost the context.

Reply:
So if we are to introduce HarborSelectionItem, I think it should still do some internal logic regarding FROM and TO. Then it would be much more similar to the content of the .map function, though without the array to map

Reply:
Here is something in between. Perhaps the most readable solution:


```

<HarborSelectionItem
  ref={harborInputSectionItemRef}
  fromOrTo={FROM}
  harbor={fromHarbor}
  disabled={false}
  onPress={() => {
    onSelect({
      fareProductTypeConfig,
      preassignedFareProduct,
    });
  }}
/>
<HarborSelectionItem
  ref={harborInputSectionItemRef}
  fromOrTo={TO}
  harbor={toHarbor}
  disabled={!fromHarbor}
  onPress={() => {
    fromHarbor &&
      onSelect({
        fromHarbor,
        fareProductTypeConfig,
        preassignedFareProduct,
      });
  }}
/>
          
          
          
--------
          
type HarborSelectionItemProps = {
  harbor?: StopPlaceFragment;
  disabled: boolean;
  onPress: () => void;
  fromOrTo: typeof FROM | typeof TO;
};

const HarborSelectionItem = forwardRef<
  TouchableOpacity,
  HarborSelectionItemProps
>(
  (
    {harbor, onPress, disabled, fromOrTo}: HarborSelectionItemProps,
    harborInputSectionItemRef,
  ) => {
    const itemRef = useRef<TouchableOpacity>(null);
    const {t} = useTranslation();
    const styles = useStyles();

    return (
      <GenericClickableSectionItem
        ref={itemRef}
        accessibilityState={{disabled}}
        accessible={true}
        accessibilityRole="button"
        accessibilityLabel={t(
          PurchaseOverviewTexts.stopPlaces.harborSelection[fromOrTo].a11yLabel(
            harbor?.name,
          ),
        )}
        accessibilityHint={
          disabled
            ? undefined
            : t(
                PurchaseOverviewTexts.stopPlaces.harborSelection[fromOrTo]
                  .a11yHint,
              )
        }
        onPress={() => {
          onPress();
          if (
            harborInputSectionItemRef &&
            typeof harborInputSectionItemRef !== 'function'
          ) {
            harborInputSectionItemRef.current = itemRef.current;
          }
        }}
        testID="selectHarborsButton"
      >
        <>
          <View style={fromOrTo === FROM ? styles.fromHarbor : styles.toHarbor}>
            <ThemeText
              color={disabled ? 'disabled' : 'secondary'}
              type="body__secondary"
              style={styles.toFromLabel}
            >
              {t(PurchaseOverviewTexts.fromToLabel[fromOrTo])}
            </ThemeText>
            <HarborLabel harbor={harbor} disabled={disabled} />
          </View>
        </>
      </GenericClickableSectionItem>
    );
  },
);
          ```